core/src/main/java/org/apache/iceberg/rest/auth/AuthConfig.java

```diff
@@ -47,7 +47,7 @@ default String scope() {
     return OAuth2Properties.CATALOG_SCOPE;
   }
 
-  @Value.Lazy
+  @Value.Default
```

Comment:
why do we want to make this change? This implies that the expiration is now settable on the Builder. However, the expiration should be determined from the actual token

Reply:
It's settable on the builder when creating a config from catalog properties:

https://github.com/apache/iceberg/blob/a705635d06a7cbc7e99eb846b87761e840773a38/core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java#L89

Reply:
Did we resolve this?  I'm not sure I fully understand what immutables is doing here.

Reply:
The only difference is that with `@Lazy` there will be no builder method generated for that attribute, effectively making it a read-only/derived property. With `@Default`, it's a separate property with its own builder method – but the same default initial value.

I think `@Default` makes more sense here, because we do have a property `TOKEN_EXPIRES_IN_MS` that users can set, so it would be optimal if the user value could be reflected here.

But I don't want to fight for this: if you prefer we can revert back to `@Lazy`, in which case the `TOKEN_EXPIRES_IN_MS` property would have to be stored somewhere else, outside of this class. No big deal.

I will leave the final word to any of you @nastra or @danielcweeks :-)

Reply:
after reviewing this portion of the code I believe this change is fine